### DISCUSSION TITLE
userprefs

### DISCUSSION THREAD

**Discussion by se7en-x230** [prasanthrangan/hyprdots#1079] [Ideas]
Any way to make the userprefs overwrite

1.  keybindings that are already set in keybindings.conf OR unbind them first in userpref before assigning new ones?
2.  defaults from keybindings.conf like 
# assign apps
$term = st
$editor = nvim
$file = st -e lf
$browser = vieb

This way I could keep all hypr configs untouched and only edit userprefs.conf

**Comment by kRHYME7** [comment]
Check the Hyprland wiki there is the ``` unbind ``` option.


A while back I wrote a script to compare 2 files then it will auto add the ``` unbind.conf ``` which effectively list and unbind the keys on the keybindings.conf  if it is use in the userprefd file. Unfortunately, the PR was turned down so  I added it to the hyprdots-ctl .


It is already on your system😉, but I did not add it for CLI. 

That is the script, you can check it first 
``` /usr/lib/hyprdots-ctl/unbind_keybindings.sh ```

![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/f778f1f6-7c2e-48b6-b9ca-510ccd6e7df4)


the file will look like this 

![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/642519b8-a767-4725-a7b8-287d02eb3d91)

Then it will be automatically source to the top most line of the userprefs 

![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/9db0b458-8046-42ec-89d3-a8830866ec51)


If you find this useful I will implement  it on the hyprdots-ctl as an official feature

**Comment by se7en-x230** [comment]
Very useful. 
I will have to look why the script seams to not find any duplicated keybindings tomorrow but it might have to do with my userpref.conf setup

![image](https://github.com/prasanthrangan/hyprdots/assets/69777713/b3be65ff-c60e-4ed5-b17b-bfd25a1dd349)

![image](https://github.com/prasanthrangan/hyprdots/assets/69777713/c20b733a-c2bb-4e81-b841-46902b6688f9)

**Reply by kRHYME7** [reply]
I'll be happy to maintain the script if someone actually uses this. 


BTW   I remember that this one had a dependency ``` hyprkeys ```

**Reply by se7en-x230** [reply]
correct i installed hyprkeys already

I would use it even so knowing to be able to source a unbind.conf file and setting that up once manually is good already.
Scripting this process is obviously always appreciated.

**Comment by se7en-x230** [comment]
Can the same be done with the default apps? 
Meaning to source a separate file or "unbind" them as not everyone uses the same defaults? Keeping the main *.configs untouched and sourcing the custom settings would be easy to keep the main configs up to date while the customs overwrite the settings.

**Comment by se7en-x230** [comment]
Can the same be done with the default apps?
$term = st
$editor = nvim
$file = st -e lf
...

That would mean keybindings.conf can always be updated to latest because userprefs.conf/unbind.conf/defaultapps.conf will overwrite the default settings.

**Reply by kRHYME7** [reply]
If you make a duplicate of the keybindings.conf with your variables at the top it will overwrite all of it

**Reply by se7en-x230** [reply]
Yes I was hoping I could either 
"unbind" and "rebind" the defaults in userpref 
or
source/overwrite the defaults from another file

**Reply by se7en-x230** [reply]
Oh I can overwrite from userpref.conf. Exellent,

Edit, nope does not work this way. 
But I can unbind the browser key and bind it to another browser

**Comment by se7en-x230** [comment]
Apart from the default apps I managed via unbind and userprefs.conf to keep the keybindings.conf untouched.
Thanks for pointing me in the right direction

![image](https://github.com/prasanthrangan/hyprdots/assets/69777713/d3956e07-879a-49e6-a268-c3832f770181)
